src/Roster.Core/Domain/Member.cs

```diff
@@ -154,8 +154,9 @@ public void ToggleAutomaticDischarge()
 
         public void Promote(RankId rankId)
         {
+            int oldRankId = rankId;
             RankId = rankId;
-            Publish(new MemberPromoted(Nickname, RankId.Id, DateTime.UtcNow));
+            Publish(new MemberPromoted(Nickname, RankId.Id, oldRankId, DateTime.UtcNow));
```

Comment:
Isn't this assigning `RankId` and `OldRankId` in `MemberPromoted` the exact same value?

Reply:
Fixed, it was a typo.